CedarJava/src/test/java/com/cedarpolicy/SharedIntegrationTests.java

```diff
@@ -353,20 +354,29 @@ private void executeJsonQueryTest(
                         query.principal == null ? Optional.empty() : Optional.of(query.principal),
                         query.action,
                         query.resource == null ? Optional.empty() : Optional.of(query.resource),
-                         Optional.of(query.context),
+                        Optional.of(query.context),
                         Optional.of(schema));
         Slice slice = new BasicSlice(policies, entities);
-        AuthorizationResponse result = assertDoesNotThrow(() -> auth.isAuthorized(authQuery, slice));
-
-        assertEquals(query.decision, result.getDecision());
-        //Errors can disagree if e.g.,
-        // <[error occurred while evaluating policy `policy0`: wrong number of arguments provided to extension function isInRange: expected 2, got 0]>
-        // but was: <[couldn't parse policy with id policy0, poorly formed: invalid syntax, expected function, found isInRange]>
-        // This skips ~48 tests. Of course, the authorization decision will be the same, even though the error may be different
-        //TODO: fix this
-        if(result.getErrors().stream().noneMatch(errorMsg -> errorMsg.contains("couldn't parse policy with id"))) {
+        
+        try {
+            AuthorizationResponse result = auth.isAuthorized(authQuery, slice);
+            assertEquals(query.decision, result.getDecision());
             assertEquals(query.errors, result.getErrors());
             assertEquals(new HashSet<>(query.reasons), result.getReasons());
+        } catch (BadRequestException e) {
+            // In case of a parse error, isAuthorized and validate will throw `BadRequestException`.
+            // A few of our corpus tests currently fall into this category because a function call
+            // (e.g., `foo()`) is considered a parse error when the function name is unrecognized.
+
+            // TODO: this current fails for 48 test cases
+            // Some of these failures are expected (due to the comment above) but the "invalid" token ones
+            // are a little odd -- it looks like some policies being sent to the authorizer aren't the same
+            // as what's in the corpus files. Some issue with unicode?
+            throw new AssertionError("parse error: " + e.getMessage());
```

Comment:
I plan to address this TODO before merging... but looking for suggestions for the best approach. What can we assert in the case where there is a parse error?